lib/cfg-tree.c

```diff
@@ -1065,6 +1066,38 @@ cfg_tree_compile(CfgTree *self)
   return TRUE;
 }
 
+static gboolean
+are_the_persist_ids_unique(const GPtrArray * const initialized_pipes)
+{
+  GHashTable * pipe_persist_ids = g_hash_table_new(g_str_hash, g_str_equal);
+  gboolean result = TRUE;
+
+  for (gint i = 0; i < initialized_pipes->len; ++i)
```

Comment:
It would be nice, but the function has `void` return type, and I would need a non-void type in order to know if was there a name collision or not.
This could be worked around with an output parameter, but that would make the code more complex.